packages/cadl-ranch/src/actions/validate-mock-apis.ts

```diff
@@ -15,6 +15,19 @@ export async function validateMockApis({ scenariosPath }: ValidateMockApisConfig
   const cadlCompiler = await importCadl(scenariosPath);
   const cadlRanchExpect = await importCadlRanchExpect(scenariosPath);
   const diagnostics = createDiagnosticReporter();
+
+  const foundUris = new Set<string>();
+  mockApis.forEach(mockApi =>
+    Object.values(mockApi.scenarios).forEach(scenario =>
+      scenario.apis.forEach(api => {
+        const key = `${api.method} ${api.uri}`;
+        foundUris.has(key)
+          ? diagnostics.reportDiagnostic({
+            message: `Uri ${api.uri} with HTTP verb ${api.method} is being defined twice.`,
```

Comment:
So there is already validation to make sure 2 scenario don't define the same URL here https://github.com/Azure/cadl-ranch/blob/7435655e82b732b3573379f76e0e613423d244e9/packages/cadl-ranch/src/scenarios-resolver.ts#L83

Making sure 2 endpoints are not the same could be useful too but then I would update this message to be 
`... is being defined twice in Mock Apis.`

Reply:
If I understand correctly, this snippet is to validate the cadl file, not the mockapi file. That is why validate-scenarios could stilll pass even there exists duplicate endpoint.

I hope the mockapi file, which is used when we start cadl-ranch server, could fail when it defines duplicate endpoints.

It would be even better if we could check the consistency between cadl file and mockapi file.

Reply:
consistency check between the scenario and implemention would  be nice but it is a little tricky as the mockapi endpoints might not exactly match the cadl (could be a parameter in the cadl but implemented as multiple endpoints in the mockapi)

Reply:
Understand. So we still might need this PR to validate the mockapi, so that next time others and I won't take so much time figure out why the ranch server cannot recognize the request.